WIP: display # of currently connected peers for the desktop client

rcullito opened the pull request:
…on the lower portion of the profile page

fixes #6567 

### Summary:
Beginnings of a PR that will increase transparency of connections for the status desktop client. Most discussion around which exact metrics should be displayed, and how, can be found in the linked issue above. 
...

### Review notes
TODO

### Testing notes 
TODO

status: ready <!-- wip -->

Commit:
Add Fastlane action to upload Google Play metadata

Signed-off-by: Goran Jovic <goranjovic@gmail.com>

mandrigin committed:
Show debug information when LES is selected (in the titlebar).

To simplify testing and debugging, there is a very simple sync indicator
(debug) for LES. This UI is not even close to being production and will
be reworked in the future.

Signed-off-by: Igor Mandrigin <i@mandrigin.ru>

goranjovic committed:
feature #6509 - validating token config against their contracts; moved token info from compile time list to app-db; fixed any discrepancies in existing info

Signed-off-by: Goran Jovic <goranjovic@gmail.com>

jakubgs committed:
stop using build-* git tags for counting build numbers

Signed-off-by: Jakub Sokołowski <jakub@status.im>

Commit:
Add support for generating Windows install program instead of zip file. Part of #6304

Signed-off-by: Pedro Pombeiro <pombeirp@users.noreply.github.com>

lukaszfryc committed:
Allow to measure battery and network consumption on real device

hetvart committed:
new test added: user can see own assets after account recovering

Signed-off-by: Oleksii Lymarenko <alexey.lymarenko@gmail.com>

dshulyak committed:
Add rendezvous nodes to configuration

Signed-off-by: Goran Jovic <goranjovic@gmail.com>

antdanchenko committed:
added runner and testrail report

Signed-off-by: Anton Danchenko <ant.danchenko@gmail.com>

hetvart committed:
new test added: check fiat value is correct for token transactions

Signed-off-by: Oleksii Lymarenko <alexey.lymarenko@gmail.com>

Commit:
[fix] align min input container height with tab menu

Signed-off-by: yenda <eric@status.im>

Commit:
[refactor] move chat.subs fns to chat.db

Signed-off-by: yenda <eric@status.im>

Commit:
[#5749] transaction history fixes

Signed-off-by: Goran Jovic <goranjovic@gmail.com>

jakubgs committed:
use a table to post links to build artifacts

Signed-off-by: Jakub Sokołowski <jakub@status.im>

Commit:
Show chat name, source contact and time in PN body. Part of #6621

- Add higher priority for 1:1 messages

Signed-off-by: Goran Jovic <goranjovic@gmail.com>

Commit:
[#6718] restyle wallet onboarding flow

Signed-off-by: Goran Jovic <goranjovic@gmail.com>

rasom committed:
[slow sign in]
fix iterating over all messages from realm db (was done for deduplication)
async loading of chats (:init-chats event)

yevh-berdnyk committed:
Fixed transactions history test

Signed-off-by: yevh-berdnyk <ie.berdnyk@gmail.com>

Commit:
display # of currently connected peers for the desktop client (on the lower portion of the profile page)

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?

Commit:
Add missing configuration for Windows desktop notifications. Part of #6728

Commit:
Fix QTROOT for clang64-based Qt

Signed-off-by: Vitaliy Vlasov <siphiuel@gmail.com>

antdanchenko committed:
swipe to delete fix

Signed-off-by: Anton Danchenko <ant.danchenko@gmail.com>

Commit:
updating cljs compiler

Signed-off-by: yenda <eric@status.im>

andytudhope committed:
Update links

Signed-off-by: Andrea Maria Piana <andrea.maria.piana@gmail.com>

FrederikBolding committed:
Changed to popups for installing and uninstalling extensions

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

cammellos committed:
Bump version to 0.9.31

Signed-off-by: Andrea Maria Piana <andrea.maria.piana@gmail.com>

hetvart committed:
recovery view: added accessibility labels for passphrase and password

Signed-off-by: Oleksii Lymarenko <alexey.lymarenko@gmail.com>

Commit:
[fix] resurrect cljs-react-navigation

Signed-off-by: yenda <eric@status.im>

hetvart committed:
new test added: check that system replaces capital letters when typing passphrase

Signed-off-by: Oleksii Lymarenko <alexey.lymarenko@gmail.com>

Commit:
Ensure no ubuntu-server process is running at startup. Fixes #6584

Signed-off-by: Pedro Pombeiro <pombeirp@users.noreply.github.com>

flexsurfer committed:
[#6736] support multi-extensions store

Signed-off-by: Andrey Shovkoplyas <motor4ik@gmail.com>

oskarth committed:
(No code): Github issue simpte template (#6444)

Commit:
introduce priority map

- in future PRs we want to reduce the expensive sort operations
on list of messages while keeping the possibility to get a message
by its ID
- priority map allow to keep a map sorted by it's value which
is what we want to do here. we want to keep the messages ordered
by clock-value

Signed-off-by: yenda <eric@status.im>

Commit:
[refactor] move contact.subs and utils.contacts to contact.db

functions defined in subs need to be reusable and testable independently
from re-frame framework. they are moved into db namespace for that purpose

Signed-off-by: yenda <eric@status.im>

antdanchenko committed:
skip tests in order not to false fail in PR

Signed-off-by: Anton Danchenko <ant.danchenko@gmail.com>

Commit:
[refactor] differentiate private and public chat and contact subs

- aliased namespace keywords are used for local subs used for caching
- synthetic :chat namespace is used for subs used elsewhere

Signed-off-by: yenda <eric@status.im>

goranjovic committed:
bug #6768 - updating enabled tokens before navigation to wallet

Signed-off-by: Goran Jovic <goranjovic@gmail.com>

Commit:
[fix] query-chat-contacts subscription params

Signed-off-by: yenda <eric@status.im>

Commit:
[fix] usernames in bold on Linux

Signed-off-by: yenda <eric@status.im>

Commit:
[fix] current chat in bold on Linux

Signed-off-by: yenda <eric@status.im>

hetvart committed:
fix faq element locator on help view

Signed-off-by: Oleksii Lymarenko <alexey.lymarenko@gmail.com>

Commit:
[#6786] restyle main wallet screen

Signed-off-by: Jakub Sokołowski <jakub@status.im>

jeluard committed:
[Fixes #6739] Introduced wallet.settings hook

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

Commit:
Reduce number of days required to consider a PR stale to 14

Signed-off-by: Pedro Pombeiro <pombeirp@users.noreply.github.com>

rasom committed:
[slow sign in] Add unviewed messages counter to chat entity.

Before we fetched ALL user-statuses with `status=received` (which means that
a message hasn't been seen), iterated them, grouped by chat and then stored
`message-ids` of these `user-statuses` in chat's `:unviewed-messages` key.

This commit introduces :unviewed-messages-count field in chat entity.
That means that there is no need to iterate `user-statuses` in order to count
a total number of unviewed messages, it is always stored along with chat.
In the rest of it, the difference is only that chat's db record should be
updated each time when unviewed messages are seen.

jakubgs committed:
fix naming of download status-go archives to avoid re-downloads

Signed-off-by: Jakub Sokołowski <jakub@status.im>

antdanchenko committed:
removed sign in after the network switch

Signed-off-by: Anton Danchenko <ant.danchenko@gmail.com>

Commit:
Lock down conan.cmake file version to avoid having upstream break our builds

Commit:
better protect transaction history polling from errors

Go blocks parse try catch blocks and turn them into event
dispatches. This captures the original intent of the code to catch
errors and terminate the current async worker execution.

Signed-off-by: yenda <eric@status.im>

hetvart committed:
new test added: check currency fiat value is correct

Signed-off-by: Oleksii Lymarenko <alexey.lymarenko@gmail.com>

jeluard committed:
[Fixes #6784] Fixed some ABI codec issues

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

jeluard committed:
[Fixes #6793] Improved hook name

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

rasom committed:
[slow sign in] faster get-unviewed-messages

jeluard committed:
Renamed hook name

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

flexsurfer committed:
[#6643] Allow chat command suggestions to select parameter

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

rasom committed:
[slow sign in] faster :get-referenced-messages

hetvart committed:
new test added: user can send/read messages in public chat

Signed-off-by: Oleksii Lymarenko <alexey.lymarenko@gmail.com>

jeluard committed:
[Fixes #6825] Do not open settings as a modal

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

Commit:
[#6815] restyle sent-transaction screen

Signed-off-by: Goran Jovic <goranjovic@gmail.com>

dshulyak committed:
Rendezvous should be false if list of nodes is empty

We can disable this validation in status-go. I am not sure why we have it.
But it will take longer.

Signed-off-by: Igor Mandrigin <i@mandrigin.ru>

chadyj commented:
Heya @rcullito how is it going?

flexsurfer committed:
[#6642] Allow adding extra key/value pair to chat command

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

rasom committed:
fix user-statuses after migration from #6722

dshulyak committed:
Add LES nodes back into list of static nodes

Signed-off-by: Igor Mandrigin <i@mandrigin.ru>

rcullito commented:
Hey @chadyj thanks for following up. To be honest, it took me a while to figure out the inner workings of the repo again and I've written up my experience [here](https://discuss.status.im/t/the-journey-to-inline-eval-in-status-react/790):  and Andy's helped turn it into a [tutorial](https://github.com/status-im/status.im/blob/feature/tutorials/source/tutorials/desktop_inline_eval.md) for new contributors .

I'm planning on resuming work on the actual feature implementation in this PR again tomorrow now that there are clear development skies once again 🌞

jeluard committed:
Fixed broken extensions HTTP events

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

Commit:
display mailserver state and peers count under advanced settings on
the desktop profile tab

rcullito commented:
closing and moving to https://github.com/status-im/status-react/pull/6865
